diff --git a/src/keys.c b/src/keys.c
index 16a4c1d..c3b89f2 100644
--- a/src/keys.c
+++ b/src/keys.c
@@ -12,6 +12,7 @@
 
 #include "structs.h"
 #include "keys.h"
+#include <ctype.h>
 
 #ifndef NO_KBD
 
diff --git a/src/misc.c b/src/misc.c
index 315f279..85d997f 100644
--- a/src/misc.c
+++ b/src/misc.c
@@ -11,6 +11,7 @@
  */
 
 #include "misc.h"
+#include <sys/wait.h>
 
 static int trapped_error_code = 0;
 static int (*old_error_handler) (Display *d, XErrorEvent *e);
