diff --git a/configure.ac b/configure.ac
index eef2f61..c84d3f9 100644
--- a/configure.ac
+++ b/configure.ac
@@ -93,6 +93,8 @@ AC_ARG_ENABLE(gcov,
   [  --enable-gcov                 enable gcov coverage ( testing only ) build],
      enable_gcov=$enableval, enable_gcov=no )
 
+  m4_ifndef([AM_GCONF_SOURCE_2], [AC_DEFUN([AM_GCONF_SOURCE_2],[AM_CONDITIONAL([GCONF_SCHEMAS_INSTALL],[false])])])
+
 AC_ARG_ENABLE(gprof,
   [  --enable-gprof                enable gprof profiler ( testing only ) build],
      enable_gprof=$enableval, enable_gprof=no )
